examples/stats/multiple_wragged_outputs.ipynb

```diff
@@ -0,0 +1,424 @@
+      "text/plain": [
+       "<Figure size 432x288 with 3 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# solve system\n",
+    "sol_12 = problem_0.evaluate(real_parameters)\n",
+    "\n",
+    "## this uses cached result from first solve under the hood\n",
+    "sol_3 = problem_1.evaluate(real_parameters)\n",
+    "\n",
+    "# overlay ODE solutions over noisy data\n",
+    "plt.figure()\n",
+    "plt.subplot(3, 1, 1)\n",
+    "plt.plot(times_12, outputs_12[:, 0], 'b')\n",
+    "plt.plot(times_12, sol_12[:, 0])\n",
+    "plt.subplot(3, 1, 2)\n",
+    "plt.plot(times_12, outputs_12[:, 1], 'g')\n",
+    "plt.plot(times_12, sol_12[:, 1])\n",
+    "plt.subplot(3, 1, 3)\n",
+    "plt.plot(times_3, outputs_3, 'r')\n",
+    "plt.plot(times_3, sol_3)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inference\n",
+    "We now describe how we can perform inference for the overall system. Here, we assume different likelihoods for each of the two output chunks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "log_likelihood_0 = pints.GaussianKnownSigmaLogLikelihood(problem_0, [noise1, noise2])\n",
+    "log_likelihood_1 = pints.GaussianLogLikelihood(problem_1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This means that each of our log-likelihoods have a different number of parameters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "5"
+      ]
+     },
+     "execution_count": 21,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "log_likelihood_0.n_parameters()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "6"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "log_likelihood_1.n_parameters()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then aggregate these into a single callable object that we can use for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "class CombinedLogLikelihood(pints.LogPDF):\n",
```

Comment:
This is exactly the kind of thing I'd want PINTS to handle for me, after I've gone to the trouble of setting up the special problem class!